agent/agent_nuclei.py

```diff
@@ -60,6 +61,39 @@ class Target:
     port: Optional[int] = None
 
 
+@dataclasses.dataclass
+class BasicCredential:
+    """Credential for form-based authentication passing password and login."""
+
+    def __init__(self, login: str, password: str) -> None:
+        self.login = login
+        self.password = password
+
+    @property
+    def basic_auth_header(self) -> str:
+        credentials = f"{self.login}:{self.password}"
+
+        encoded_credentials = base64.b64encode(credentials.encode("utf-8")).decode(
+            "utf-8"
+        )
+
+        auth_header = f"Authorization: Basic {encoded_credentials}"
+        return auth_header
+
+
+def build_basic_credential_from_message(message: m.Message) -> BasicCredential | None:
+    basic_credential = message.data.get("basic_credential")
+    if basic_credential is None:
+        return None
+    login = basic_credential.get("login")
+    if login is None:
+        return None
+    password = basic_credential.get("password", "")
```

Comment:
You should remove the "" so we can get none
```suggestion
    password = basic_credential.get("password")
```

Reply:
Alright, good catch. Thanks